apps/betterangels-backend/clients/schema.py

```diff
@@ -130,46 +192,68 @@ def _validate_hmis_profiles(hmis_profiles: list[dict[str, Any]]) -> list[dict[st
             )
             .exists()
         ):
-            errors.append(_build_error("hmisProfiles", f"{idx}__hmisId", ErrorMessageEnum.HMIS_ID_IN_USE.name))
+            errors.append(
+                {
+                    "field": "hmisProfiles",
+                    "location": f"{idx}__hmisId",
+                    "errorCode": ErrorCodeEnum.HMIS_ID_IN_USE.name,
+                }
+            )
 
     return errors
 
 
-def _validate_contacts(contacts: list[dict[str, Any]]) -> list[dict[str, Any]]:
+def validate_contacts(contacts: list[dict[str, Any]]) -> list[dict[str, Any]]:
     errors = []
 
     for idx, contact in enumerate(contacts):
+        if not contact.get("phone_number"):
+            continue
+
         try:
-            phonenumber_field.validators.validate_international_phonenumber(contact["phone_number"])
+            validate_international_phonenumber(contact["phone_number"])
         except ValidationError:
-            errors.append(_build_error("contacts", f"{idx}__phoneNumber", ErrorMessageEnum.INVALID_PHONE_NUMBER.name))
+            errors.append(
+                {
+                    "field": "contacts",
+                    "location": f"{idx}__phoneNumber",
+                    "errorCode": ErrorCodeEnum.PHONE_NUMBER_INVALID.name,
+                }
+            )
 
     return errors
 
 
-def _validate_client_profile_data(data: dict) -> None:
+def validate_client_profile_data(data: dict) -> dict[Any, Any]:
     """Validates the data for creating or updating a client profile."""
-    errors = []
+    errors: list = []
+
+    user = None
+
+    if value_is_set(data.get("id")):
+        user = User.objects.filter(client_profile__id=data["id"]).first()
 
-    if data["user"] is not strawberry.UNSET:
-        user_id = data["user"].get("id", None)
-        user = User.objects.filter(id=user_id).first() if user_id else None
+    if user_data := data.get("user"):
+        errors += validate_client_name(user_data, user_data.get("nickname"), user)
```

Comment:
i have a slight preference for putting all of these functions into a validator class, like:
```
class ClientProfileValidator:
    def __init__(self, data):
        self.data = data
        self.errors = []
    def validate_client_profile_data(data: dict) -> None:
        if ...:
            self.validate_...

    ...
```


then all of these validate functions can just append to self.errors directly. instead of the `errors +=` pattern, but lemme know what you think.